packages/ui/Media/Media.js

```diff
@@ -124,12 +206,28 @@ class Media extends React.Component {
 
 	componentDidMount () {
 		this.attachCustomMediaEvents();
+
+		this.autoLoad();
+	}
+
+	componentDidUpdate ({source: prevSource}) {
+		const {source} = this.props;
+
+		if (getKeyFromSource(source) !== getKeyFromSource(prevSource)) {
+			this.autoLoad();
+		}
 	}
 
 	componentWillUnmount () {
 		this.detachCustomMediaEvents();
 	}
 
+	autoLoad () {
+		if (this.props.noAutoLoad) return;
```

Comment:
Good catch! It prevents *reloading* but not loading. At best, it's poorly named and should only be invoked in `componentDidUpdate`. I'll make a change.

Reply:
This could be cruft from earlier refactoring. In testing in Chrome, I've found that the "auto loading" doesn't have any effect. In the usage by `Video`, `componentDidUpdate` never fires with a changed source because the `key` generated for each `Media` component ensures that a change in source results in a new component.

For now, I've removed this logic but will need to verify on the TV hardware to ensure that there aren't regressions.